### DISCUSSION TITLE
Default Generic Type Parameter Value

### DISCUSSION THREAD

**Discussion by metasong** [dotnet/csharplang#7308] [Language Ideas]
It would be convenient to define the default value for the type parameter:
```csharp
public record struct Controllable<Unit, T = double>(Unit unit)  {T Value;}
```
so we can directly create `var goal = Controllable(Length.Meter)`, no need to do it like this `Controllable<Length, double>(Length.Meter)`

**Comment by 333fred** [comment]
See https://github.com/dotnet/csharplang/issues/1328 and https://github.com/dotnet/csharplang/issues/5556.
